aws/table_aws_api_gatewayv2_integration.go

```diff
@@ -0,0 +1,274 @@
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "request_templates",
+				Description: "Represents a map of Velocity templates that are applied on the request payload based on the value of the Content-Type header sent by the client. The content type value is the key in this map, and the template (as a String) is the value. Supported only for WebSocket APIs.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "response_parameters",
+				Description: "API requests and responses (https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-parameter-mapping.html).",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tls_config",
+				Description: "The TLS configuration for a private integration. If you specify a TLS configuration, private integration traffic uses the HTTPS protocol. Supported only for HTTP APIs.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard columns
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAPIGatewayV2IntegrationAkas,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("IntegrationId"),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAPIGatewayV2Integrations(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAPIGatewayV2Integrations", "AWS_REGION", region)
+	api := h.Item.(*apigatewayv2.Api)
+
+	// Create Session
+	svc, err := APIGatewayV2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	pagesLeft := true
+	params := &apigatewayv2.GetIntegrationsInput{
+		ApiId: api.ApiId,
+	}
+
+	for pagesLeft {
+		result, err := svc.GetIntegrations(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, integration := range result.Items {
+			d.StreamLeafListItem(ctx, integrationInfo{*integration, *api.ApiId})
+		}
+
+		if result.NextToken != nil {
+			pagesLeft = true
+			params.NextToken = result.NextToken
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAPIGatewayV2Integration(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAPIGatewayV2Integration")
+	api := d.KeyColumnQuals["api_id"].GetStringValue()
+	key := d.KeyColumnQuals["integration_id"].GetStringValue()
```

Comment:
Can we add this block after creating the session? As we don't need this variables, if `APIGatewayV2Service` returns any error.